examples/advanced_operations/add_responsive_search_ad_full.py

```diff
@@ -0,0 +1,541 @@
+#!/usr/bin/env python
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This example shows how to create a complete Responsive Search ad.
+Includes creation of: budget, campaign, ad group, ad group ad,
+keywords, geo targeting, and image extensions.
+
+More details on Responsive Search ads can be found here:
+https://support.google.com/google-ads/answer/7684791
+"""
+
+import argparse
+import sys
+import uuid
+import requests
+
+from google.ads.googleads.client import GoogleAdsClient
+from google.ads.googleads.errors import GoogleAdsException
+
+
+# Keywords from user.
+keyword_text_exact_1 = "example of exact match"
+keyword_text_phrase_1 = "example of phrase match"
+keyword_text_broad_1 = "example of broad match"
+
+# Geo targeting from user.
+location_1 = "Buenos aires"
+location_2 = "San Isidro"
+location_3 = "Mar del Plata"
+
+# LOCALE and COUNTRY_CODE are used for geo targeting.
+# Locale is using ISO 639-1 format. If an invalid locale is given,
+# 'es' is used by default.
+LOCALE = "es"
+
+# A list of country codes can be referenced here:
+# https://developers.google.com/google-ads/api/reference/data/geotargets
+COUNTRY_CODE = "AR"
+
+# To add image extensions, the account has to follow these requirements:
+# https://support.google.com/google-ads/answer/9566341
+# If the account does not meet the requirements, set below variable to False.
+add_image_extension = True
+
+'''
+Support functions
+'''
+def handle_googleads_exception(exception):
+    print(
+        f'Request with ID "{exception.request_id}" failed with status '
+        f'"{exception.error.code().name}" and includes the following errors:'
+    )
+    for error in exception.failure.errors:
+        print(f'\tError with message "{error.message}".')
+        if error.location:
+            for field_path_element in error.location.field_path_elements:
+                print(f"\t\tOn field: {field_path_element.field_name}")
+    sys.exit(1)
+
+def create_ad_text_asset(client, text, pinned_field=None):
+    """Create an AdTextAsset."""
+    ad_text_asset = client.get_type("AdTextAsset")
+    ad_text_asset.text = text
+    if pinned_field:
+        ad_text_asset.pinned_field = pinned_field
+    return ad_text_asset
+
+'''
+6 steps to create a Responsive Search Ad (RSA).
+'''
+
+def main(
```

Comment:
Can you break this method into smaller methods where each is responsible for creating a single resource? Specifically the body in `main` should look something like:

```
def main(...):
    _create_campaign_budget(...)
    _create_campaign(...)
    _create_ad_group(...)
```

The smart campaign example does a good job of this in order to make the chunks of code more discernable for the reader.